### ISSUE TITLE
How to make light yolo

### ISSUE THREAD

**Issue by Peter0452** [ultralytics/ultralytics#16919]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

![image](https://github.com/user-attachments/assets/c5bc624d-fab2-4e92-9eb1-3bd0082e74ce)

Hi there, 

I have been trying to make a light yolo with as fewest number of parameters and GFLOPs as possible to do object detection (robots). However, I dont understand why with around 50k parameters, this model takes up to 1.1 GFLOPs. I thought it should only consume like around 0.05 GLOPs when I only have a few cnn layers. 

Any insight would be appreciated!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Peter0452, thank you for your interest in Ultralytics 🚀!

If you're aiming to create a lighter YOLO model, I recommend starting by reviewing our [Model Architecture Docs](https://docs.ultralytics.com/models/architecture/) to ensure your design aligns with best practices.

Since your concern involves model efficiency, you'll want to verify the calculations for parameters and GFLOPs. Ensuring the model layers and dimensions are optimized can help achieve the desired performance. If you think there might be a bug, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to allow us to look into it more effectively. 🐛

For additional resources:
- Visit our [Docs](https://docs.ultralytics.com) for more usage examples.
- Join our community on [Discord](https://ultralytics.com/discord) 🎧 for real-time chats or [Discourse](https://community.ultralytics.com) for detailed discussions.

## Upgrade

Make sure you're using the latest version of `ultralytics` to avoid issues:

```bash
pip install -U ultralytics
```

## Environments

Ensure your environment is correctly set up. You can use environments like:
- **Google Colab**: <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>
- **Docker Image**: See our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/).

This is an automated response. An Ultralytics engineer will assist you soon to provide additional insight 🌟.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. This helps verify correct operation on all platforms.

**Comment by Y-T-G** [comment]
The profiling runs at 640x640 imgsz. You have to reduce the `imgsz` to reduce GFLOPs.

**Comment by Peter0452** [comment]
'' model.train(data="E:/Research.../6k rb images/data.yaml", epochs=70, imgsz=192, cls=0) ''

No, this happens even when I set the image size to 192 (1.1 GFLOPs).

**Comment by Y-T-G** [comment]
Run `model.model.info(imgsz=192)`

**Comment by Peter0452** [comment]
I run your command, and it shows me this.
""WARNING ⚠️ no model scale passed. Assuming scale='n'.
modifiedYOLO summary (fused): 43 layers, 47,817 parameters, 47,801 gradients, 0.1 GFLOPs""

However, when I run model train with imgsz of 192, the result is as followed:
""modifiedYOLO summary (fused): 43 layers, 47,947 parameters, 47,931 gradients, 1.1 GFLOPs""

How should I understand this difference?

**Comment by Y-T-G** [comment]
During training, the log is based on imgsz 640.

**Comment by Peter0452** [comment]
So, during training it costs 1.1 GFLOPs, and despite what the summary says, during inference time, if I set the image to 192, it would only cost 0.1 GFLOPs, right?

**Comment by Y-T-G** [comment]
Yes

**Comment by Peter0452** [comment]
Thanks

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.
